### PR TITLE
tweaks to vscode debugging configs in template example

### PR DESCRIPTION
**Pull request by prosper00**
The .vscode configs in the template project seemed a bit old and didn't work for me. The changes I made fix some intellisense complaints with ch32v003fun.c syntax, as well as enable native vscode debugging (i.e. platformio not required).  Also adds a register viewer, with the necessary .SVD file snagged from MounRiver

Requires the cortex-debug extension (or, at least, the 'peripheral viewer' extension that's pulled in as a dependency by cortex-debug.  Also recommend the makefile tools extension to help configure intellisense with the correct defines and include paths.

### CODE REVIEW

**Comment by cnlohr** [comment]
@maxgerhardt  - in case it makes sense to ping you on github instead, I was hoping you may have some experience here?  I just haven't used VS so I don't want to blanketly trust this PR.

File: examples/template/.vscode/launch.json [hunk 1]
```diff
@@ -1,39 +1,47 @@
...
+			"deploySteps": [ // 'make ...gdbserver' doesn't seem to work here. The Makefile calls 'minichlink -baG'... Needs -aG. Easier to add minichlink as a seperate step below
+				{
+					"type": "shell", //isn't there some way to call a task from tasks.json?
+					"command": "make --directory=${workspaceFolder} -j 1 closechlink flash",
+				},
 				{
 					"type": "shell",
-					"continueOn": "GDBServer",
-					"command": "make --directory=${workspaceFolder} closechlink flash gdbserver"
+					"command": "${workspaceFolder}/minichlink/minichlink -aG",
```
Comment (maxgerhardt) [inline examples/template/.vscode/launch.json:21]: So the above change in `program` as `${workspaceFolder}/template.elf` indicates that `${workspaceFolder}` is pointing to the actual example folder, not the `ch32v003fun` "root" directory itself. Then how can minichlink exist at `${workspaceFolder}/minichlink/minichlink -aG` when you rewrote this command?

Maybe these template files have been written with the expectation that they be copied into the root of the ch32v003fun folder?

Comment (prosper00) [reply examples/template/.vscode/launch.json:21]: oh, yes, you are correct. The paths are still set to my own project directory structure, not the 'fun examples structure. Will update when I get a sec

**Comment by cnlohr** [comment]
Looks good except for the above note -- but also I am wondering, what is the chonker svd file for?

**Comment by cnlohr** [comment]
@prosper00 any chance you could fix the conflict?

**Comment by prosper00** [comment]
I fixed the paths in this commit: https://github.com/cnlohr/ch32v003fun/pull/298/commits/0ede87fee66eeb773a94e3186d6fe8050bed4751

Is there more you 'd like to see?

File: examples/template/.vscode/launch.json (hunk 1 above)
Comment (maxgerhardt) [reply examples/template/.vscode/launch.json]: Thanks, will recheck project in VSCode to make sure everything's fine.

**Comment by cnlohr** [comment]
> I fixed the paths in this commit: [0ede87f](https://github.com/cnlohr/ch32v003fun/commit/0ede87fee66eeb773a94e3186d6fe8050bed4751)
> 
> Is there more you 'd like to see?

You need to remove the merge conflict (actually introduce within that commit) https://github.com/cnlohr/ch32v003fun/commit/0ede87fee66eeb773a94e3186d6fe8050bed4751

You can fix it on your current commit as to not mess with the git history.

**Comment by prosper00** [comment]
Ah, got it. Ok, should be good now

**Comment by cnlohr** [comment]
eyyyyyy

**Comment by cnlohr** [comment]
@prosper00 You ok with a merge now? 
@maxgerhardt Good with merge now?

**Comment by prosper00** [comment]
> @prosper00 You ok with a merge now? 
> @maxgerhardt Good with merge now?

good w/me

**Pull request merged**